doc/md/fundamentals/2-types/4-tuples.md

```diff
@@ -91,18 +60,46 @@ Tuples can be stored in arrays or other data structures. Tuples can be construct
 let users : [(Text, Nat)] = [("Motoko", 25), ("Ghost", 30)];
 ```
 
-This structure efficiently represents a collection of key-value pairs without requiring a dedicated [record](https://internetcomputer.org/docs/motoko/fundamentals/types/records) type. 
+This structure efficiently represents a collection of key-value pairs without requiring a dedicated [record](https://internetcomputer.org/docs/motoko/fundamentals/types/records) type.
 
-## Extracting values using switch
+## Pattern matching on tuples
 
-As well as the dot notation, tuple can be decomposed using tuple patterns. Combined with `let` or `switch`, this let you access the components of a tuple by simple pattern matching. 
+As well as the dot notation, tuple can be decomposed using tuple patterns. Combined with `let` or `switch`, this let you access the components of a tuple by simple pattern matching.
 
 ```motoko no-repl
 let users : [(Text, Nat)] = [("Motoko", 25), ("Ghost", 30)];  
 
 let (firstUserName, _) = users[0] // "Motoko" 
 ```
 
-The array `users` contains tuples, where each tuple represents a user with a [`Text`](https://internetcomputer.org/docs/motoko/base/Text) name and a [`Nat`](https://internetcomputer.org/docs/motoko/base/Nat) age. `users[0]` retrieves the first tuple in the array. The `let` pattern extracts only the name while ignoring the second element using the wildcard pattern (`_`). 
+The array `users` contains tuples, where each tuple represents a user with a [`Text`](https://internetcomputer.org/docs/motoko/base/Text) name and a [`Nat`](https://internetcomputer.org/docs/motoko/base/Nat) age. `users[0]` retrieves the first tuple in the array. The `let` pattern extracts only the name while ignoring the second element using the wildcard pattern (`_`).
+
+## Nesting tuples
+
+Tuples can be used to represent coordinate pairs, allowing for structured calculations such as finding the gradient of a line. The gradient (or slope) between two points $(x_1, y_1)$ and $(x_2, y_2)$ is calculated using the formula:
+
+$$
+m = \frac{y_2 - y_1}{x_2 - x_1}
+$$
+
+Using nested tuples, this can be implemented in Motoko as follows:
+
+```motoko
+type Point = (Float, Float);
+// Line is a tuple of points, that is, a nested tuple
+type Line = (Point, Point);
```

Comment:
```suggestion
  type Point = (Float, Float);
  // Line is a tuple of points, that is, a nested tuple
  type Line = (Point, Point);
```
The indentation is inconsistent. Either indent the first three lines (as suggested) or undent the remaining ones.